packages/dbos-compiler/cli.ts

```diff
@@ -0,0 +1,120 @@
+  await fsp.mkdir(outDir, { recursive: true });
+
+  const createFile = await fsp.open(path.join(outDir, "create.sql"), "w");
+  try {
+    const append = async (sql: string) => { await createFile.write(sql); };
+    await generateCreate(append, result, appVersion);
+  } finally {
+    await createFile.close();
+  }
+
+  const dropFile = await fsp.open(path.join(outDir, "drop.sql"), "w");
+  try {
+    const append = async (sql: string) => { await dropFile.write(sql); };
+    await generateDrop(append, result, appVersion);
+  } finally {
+    await createFile.close();
+  }
+}
+
+async function deployToDatabase(config: pg.ClientConfig, result: CompileResult, appVersion?: string | boolean) {
+  const client = new pg.default.Client(config);
+  try {
+    await client.connect();
+    console.log(`Deploying to database: ${client.host}:${client.port ?? 5432}/${client.database}`);
+    const append = async (sql: string) => { await client.query(sql); };
+    await generateCreate(append, result, appVersion);
+  } finally {
+    await client.end();
+  }
+}
+
+function getPackageVersion(): string {
+  const __dirname = import.meta.dirname;
+  const packageJsonPath = path.join(__dirname, "..", "package.json");
+  const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, "utf-8")) as { version: string };
+  return packageJson.version;
+}
+
+const program = new Command();
+
+program
+  .name("dbosc")
+  .description('DBOS Stored Procedure compiler')
+  .version(getPackageVersion());
+
+interface CompileOptions {
+  outDir?: string,
+  appVersion?: string | boolean,
+}
+
+function verifyTsConfigPath(tsconfigPath: string | undefined, cwd?: string): string | undefined {
+  cwd = cwd ?? process.cwd();
+
+  if (!tsconfigPath) {
+    tsconfigPath = path.join(cwd, "tsconfig.json");
+  }
+
+  if (!fs.existsSync(tsconfigPath)) {
+    console.error(`tsconfig file not found: ${tsconfigPath}`);
+    return undefined;
+  }
+
+  return tsconfigPath;
+}
+
+program
+  // FYI, commander package doesn't seem to handle app version options correctly in deploy subcommand if compile isn't also a subcommand
+  .command("compile")
+  .argument('[tsconfigPath]', 'path to tsconfig.json')
+  .option('-o, --out <string>', 'path to output folder')
+  .option('--app-version <string>', 'override DBOS__APPVERSION environment variable')
+  .option('--no-app-version', 'ignore DBOS__APPVERSION environment variable')
+  .action(async (tsconfigPath: string | undefined, options: CompileOptions) => {
+    tsconfigPath = verifyTsConfigPath(tsconfigPath);
+    if (tsconfigPath) {
+      const compileResult = compile(tsconfigPath);
+      if (compileResult) {
+        const outDir = options.outDir ?? process.cwd();
+        await emitSqlFiles(outDir, compileResult, options.appVersion);
+      }
+    }
+  });
+
+interface DeployOptions {
+  appDir?: string;
+  appVersion?: string | boolean;
+}
+
+program
```

Comment:
It might be useful to have a "delete" command to undo the deployment (with an optional app version).

Reply:
I called the command "drop"